app/src/main/java/org/wikipedia/feed/image/FeaturedImageCardView.java

```diff
@@ -29,43 +31,63 @@
         void onFeaturedImageSelected(@NonNull FeaturedImageCard card);
     }
 
+    @BindView(R.id.view_featured_image_card_content_container) View containerView;
     @BindView(R.id.view_featured_image_card_header) CardHeaderView headerView;
-    @BindView(R.id.view_featured_image_card_footer) ActionFooterView footerView;
-    @BindView(R.id.view_featured_image_card_image) FaceAndColorDetectImageView imageView;
-    @BindView(R.id.featured_image_description_Text) TextView descriptionView;
+    @BindView(R.id.view_featured_image_card_image_placeholder) FrameLayout imageViewPlaceholder;
+    @BindView(R.id.view_featured_image_card_image) ImageView imageView;
+    @BindView(R.id.view_featured_image_card_image_description) TextView descriptionView;
+    @BindView(R.id.view_featured_image_card_download_button) MaterialButton downloadButton;
+    @BindView(R.id.view_featured_image_card_share_button) MaterialButton shareButton;
+    boolean imageHasLoaded;
 
     public FeaturedImageCardView(Context context) {
         super(context);
         inflate(getContext(), R.layout.view_card_featured_image, this);
         ButterKnife.bind(this);
-        ImageZoomHelper.setViewZoomable(imageView);
     }
 
     @Override public void setCard(@NonNull FeaturedImageCard card) {
         super.setCard(card);
-        // TODO: superimpose text onto image thumb
-        image(card.image());
+        imageHasLoaded = false;
+        image(card.baseImage());
         description(defaultString(card.description()));  //Can check language before doing this if we want
         header(card);
-        footer();
-        onClickListener(new CardClickListener());
+        setClickListeners();
     }
 
     @Override public void setCallback(@Nullable FeedAdapter.Callback callback) {
         super.setCallback(callback);
         headerView.setCallback(callback);
     }
 
-    private void image(@NonNull Uri uri) {
-        imageView.loadImage(uri);
+    private void image(@NonNull FeaturedImage image) {
+        if (containerView.getWidth() == 0) {
+            // The getWidth() from containerView will return zero in the most beginning one or two cards.
+            // The imageHasLoaded is to avoid keeping setting up the image.
+            containerView.getViewTreeObserver().addOnGlobalLayoutListener(() -> loadImage(image));
```

Comment:
Can we just do a `post()` here, and not keep track of whether `imageHasLoaded`?

Reply:
You are correct! That's a proper way of doing it.